pgzx-update-0.2.0.mdx

```diff
@@ -0,0 +1,411 @@
+    }
+}
+```
+
+Once our `ereport` is compiled it behaves exactly like the C macros (minus us
+using the zig string formatting functionality). The `errstart` function checks
+if the message will be emitted and sets up the context in the error stack. The
+given options update the active error context and `errfinish` finally emits the
+error message. Depending on the log level `errfinish` might also execute a
+longjump or eveen abort the current process.
+
+Zig structs are awesome in so many ways. One rarely needs builders, let alone
+the functional options pattern. Yet it seems it is a viable pattern for logging
+and tracing APIs that are supposed to have low overhead when logging is
+disabled.
+
+## Zig update
+
+Zig 0.13 is the latest stable release, but the Zig toolchain, including the
+compiler, build system, and standard library, is still in development and
+breaking changes do happen every now and then. For this reason we currently do
+not focus on a specific Zig version, but try to follow the latest Zig
+changes in the master branch.
+
+The Nix based development shell uses
+[zig-overlay](https://github.com/mitchellh/zig-overlay) in conjunction with the
+`flake.lock` file to pin the zig toolchain version to a recent commit ID.
+
+The dependency is updated by us every so often and we try to test and fix
+breaking changes when updating the toolchain version. We highly recommend to
+use the projects development shell when testing the example extensions provided,
+otherwise you might have problems compiling the extensions if you use an older
+or newer Zig toolchain.
+
+We understand not everyone is keen to install Nix locally. For getting to know
+the environment you can build and run a development shell in a local docker
+container. Use `./dev/docker/build.sh` to build the container and
+`./dev/docker/run.sh` to start the dockerized development shell. Once you are
+inside the containers development shell you can use `./ci/setup.sh` and
+`./ci/run.sh` to build and run the projects tests.
+
+## Foreign Data Wrapper
+
+We initiated support for writing Foreign Data Wrappers
+([FDW](https://www.postgresql.org/docs/current/fdwhandler.html)) in Zig. The
+support is still very rudimentary and you still have to write C-style functions in
+Zig. We plan to provide a more Zig idiomatic API in the future.
+
+When writing an FDW you export 2 functions, the handler function that sets up a
+list of callbacks for the FDW framework and a validator function that verifies
+your FDWs options when users configure the server, user mapping, or foreign
+table. The `pgzx.fdw` modules provides some helpers to declaratively define the
+options your foreign data wrapper supports.
+
+The options API also allows you to combine multiple configurations in case you
+read a list of supported options from a library that you use for your
+implementation.
+
+This example uses reads the available options from libpq into a global table:
```

Comment:
```suggestion
This example reads the available options from libpq into a global table:
```

Reply:
Wasn't sure if this was meant to be either, or if it should read "uses and reads"?

Reply:
This sentence does not look "complete". Looks I got lost in code when I started the sentence. Will rewrite.

Reply:
OK :) The revised sentence works for me!